include/boost/multiprecision/cpp_dec_float.hpp

```diff
@@ -1564,15 +1564,7 @@ double cpp_dec_float<Digits10, ExponentType, Allocator>::extract_double() const
                      : -std::numeric_limits<double>::infinity());
    }
 
-   std::stringstream ss;
-   ss.imbue(std::locale::classic());
-
-   ss << str(std::numeric_limits<double>::digits10 + (2 + 1), std::ios_base::scientific);
-
-   double d;
-   ss >> d;
-
-   return d;
+   return std::strtod(str(std::numeric_limits<double>::digits10 + (2 + 1), std::ios_base::scientific).c_str(), nullptr);
```

Comment:
This is a pain in the butt, but can you check that this doesn't re-awaken this issue: https://github.com/boostorg/multiprecision/issues/167.  The problem is that the global locale can sometimes interpret strings in mighty weird ways :(